ReactWindows/ReactNative.Shared/UIManager/ViewKeyedDictionary.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+#if WINDOWS_UWP
+using Windows.UI.Xaml;
+using ReactNative.Bridge;
+#else
+using System.Windows;
+#endif
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// A dictionary that has <see cref="UIElement"/>'s as keys.
+    /// If a method has a <see cref="UIElement"/> parameter it must be called
+    /// on the thread the <see cref="UIElement"/> has affinity to. With this assumption
+    /// the dictionary is thread safe.
+    /// </summary>
+    /// <remarks>
+    /// This class is designed to be used as a storage of custom state for a view.
+    /// This could be needed when a view is sealed but a custom state is needed to implement
+    /// a React property in a <see cref="ViewManager{TFrameworkElement, TReactShadowNode}"/>.
+    /// The class is thread safe if any method that has a <see cref="UIElement"/> as parameter
+    /// is called on a thread it has affinity to. Thread safety is important since there may exist
+    /// multiple UI dispatchers (due to multiple windows) that use the same instance of a
+    /// <see cref="ViewManager{TFrameworkElement, TReactShadowNode}"/>.
+    /// </remarks>
+    /// <typeparam name="TKey"></typeparam>
+    /// <typeparam name="TValue"></typeparam>
+    public class ViewKeyedDictionary<TKey, TValue> where TKey : DependencyObject
+    {
+        private ConcurrentDictionary<TKey, TValue> _dictionary = new ConcurrentDictionary<TKey, TValue>();
+
+        /// <summary>Attempts to get the value associated with the specified key.</summary>
+        /// <param name="key">The key of the value to get.</param>
+        /// <param name="value">When this method returns, contains the object that has
+        ///     the specified key, or the default value of the type if the operation failed.</param>
+        /// <returns>true if the key was found otherwise, false.</returns>
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+#if WINDOWS_UWP
+            DispatcherHelpers.AssertOnDispatcher(key);
```

Comment:
I changed a little. Now we assert for both UWP/WPF, but we use a simple Dictionary in WPF (rather than a ConcurrentDictionary)

---
In reply to: [227617124](https://github.com/Microsoft/react-native-windows/pull/2001#discussion_r227617124) [](ancestors = 227617124)